src/script/media/CanvasMediaStreamMixer.ts

```diff
@@ -0,0 +1,317 @@
+        console.error('Error in mixFrames:', error);
+      }
+
+      this.animationFrame = requestAnimationFrame(mixFrames);
+    };
+
+    this.animationFrame = requestAnimationFrame(mixFrames);
+  }
+
+  async togglePictureInPicture(): Promise<void> {
+    if (!document.pictureInPictureEnabled || !window.documentPictureInPicture) {
+      console.warn('Picture-in-Picture not supported');
+      return;
+    }
+
+    try {
+      if (window.documentPictureInPicture.window) {
+        window.documentPictureInPicture.window.close();
+        this.resetPipState();
+        return;
+      }
+
+      const pipWindow = await window.documentPictureInPicture.requestWindow({
+        width: PIP_WINDOW_WIDTH,
+        height: PIP_WINDOW_HEIGHT,
+      });
+
+      this.setupPipWindow(pipWindow);
+    } catch (error) {
+      console.error('Failed to enter Picture-in-Picture mode:', error);
+      this.resetPipState();
+    }
+  }
+
+  private setupPipWindow(pipWindow: Window) {
+    pipWindow.document.body.style.cssText = `
+      margin: 0;
+      padding: 0;
+      overflow: hidden;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      background: black;
+    `;
+
+    const video = document.getElementById(PIP_VIDEO_ID) as HTMLVideoElement;
+    if (!video) {
+      throw new Error('Camera video element not found');
+    }
+
+    video.style.cssText = `
+      width: 100%;
+      height: 100%;
+      object-fit: contain;
+      display: block;
+    `;
+
+    pipWindow.document.body.appendChild(video);
+    this.isPipActive = true;
+
+    let lastUpdate = 0;
+
+    const updatePosition = () => {
+      const now = performance.now();
+      if (now - lastUpdate < POSITION_THROTTLE) {
+        return;
+      }
+
+      this.targetOffsetX = pipWindow.screenX - window.screenX;
+      this.targetOffsetY = pipWindow.screenY - window.screenY;
+      lastUpdate = now;
+    };
+
+    // Use setInterval instead of RAF for position updates
+    const positionInterval = setInterval(updatePosition, POSITION_THROTTLE);
+
+    pipWindow.addEventListener('pagehide', () => {
+      clearInterval(positionInterval);
+      this.resetPipState();
+
+      const videoContainer = document.getElementById('videoContainer');
+      if (video && videoContainer && !videoContainer.contains(video)) {
+        videoContainer.appendChild(video);
+      }
+    });
+  }
+
+  private resetPipState() {
+    this.isPipActive = false;
+    this.smallOffsetX = 0;
+    this.smallOffsetY = 0;
+    this.targetOffsetX = 0;
+    this.targetOffsetY = 0;
+  }
+
+  releaseStreams(): void {
+    if (this.animationFrame) {
+      cancelAnimationFrame(this.animationFrame);
+      this.animationFrame = null;
+    }
+
+    if (window.documentPictureInPicture?.window) {
+      window.documentPictureInPicture.window.close();
+    }
+
+    [this.screenVideo, this.cameraVideo].forEach(video => {
+      if (video) {
+        const tracks = video.srcObject as MediaStream;
```

Comment:
Theoretically that's fine, because the tracks are singletons... but it's cleaner to keep the track reference and end it and then remove the HTML element. but ok.